packages/core/test/acceptance/di_spec.ts

```diff
@@ -1919,6 +1919,128 @@ describe('di', () => {
              });
         });
       });
+
+      onlyInIvy('Ivy support InjectFlags in Injector')
+          .it('should support Optional flag in Injector', () => {
+            @Directive({selector: '[dirA]'})
+            class DirectiveA {
+              dirB = this.injector.get(DirectiveB, null, InjectFlags.Optional);
+              constructor(public injector: Injector) {}
+            }
+            @Component({template: '<div dirA></div>'})
+            class MyComp {
+              @ViewChild(DirectiveA) dirA!: DirectiveA;
+            }
+
+            TestBed.configureTestingModule({declarations: [DirectiveA, DirectiveB, MyComp]});
+            const fixture = TestBed.createComponent(MyComp);
+            fixture.detectChanges();
+
+            const dirA = fixture.componentInstance.dirA;
+            expect(dirA.dirB).toBeNull();
+          });
+
+      onlyInIvy('Ivy support InjectFlags in Injector')
+          .it('should support SkipSelf flag in Injector', () => {
+            @Component({
+              selector: 'parent',
+              template: '<child></child>',
+              providers: [{
+                provide: 'token',
+                useValue: 'PARENT',
+              }]
+            })
+            class ParentComponent {
+            }
+
+            @Component({
+              selector: 'child',
+              template: '...',
+              providers: [{
+                provide: 'token',
+                useValue: 'CHILD',
+              }]
+            })
+            class ChildComponent {
+              constructor(public injector: Injector, @SkipSelf() public parentInjector: Injector) {}
```

Comment:
I'd like to propose to refactor the code (in this and other tests) to do the following:
- inject a token via constructor params (using the necessary decorators, such as `@SkipSelf`, `@Optional`, etc)
- inject the same token via `injector.get` call
- compare results in test code

Here is a quick example:
```suggestion
              constructor(injector: Injector, @Inject('token') @SkipSelf() public tokenViaConstructor: string) {
                this.tokenViaInjector = injector.get('token', null, InjectFlags.SkipSelf);
              }
              
              // ... later in the test ...
              expect(childComponent.tokenViaConstructor).toBe('CHILD');
              expect(childComponent.tokenViaConstructor).toBe(childComponent.tokenViaInjector);
```

I believe that'd make tests more readable and would actually verify that constructor-based and `injector.get`-based behavior is the same.